ipfx/data_set_features.py

```diff
@@ -207,49 +313,35 @@ def extract_cell_features(data_set,
     ramp_start, ramp_dur, _, _, _ = stf.get_stim_characteristics(ramp_sweeps.sweeps[0].i, ramp_sweeps.sweeps[0].t)
 
     ramp_spx, ramp_spfx = extractors_for_sweeps(ramp_sweeps,
-                                                start = ramp_start,
+                                                start=ramp_start,
                                                 **detection_parameters(data_set.RAMP))
     ramp_an = spa.RampAnalysis(ramp_spx, ramp_spfx)
     ramp_features = ramp_an.analyze(ramp_sweeps)
-    cell_features["ramps"] = ramp_an.as_dict(ramp_features, [dict(sweep_number=sn) for sn in ramp_sweep_numbers ])
-
-    return cell_features
-
-
-def select_subthreshold_min_amplitude(stim_amps, decimals=0):
-    """Find the min delta between amplitudes of coarse long square sweeps.  Includes failed sweeps.
-
-    Parameters
-    ----------
-    stim_amps: list of stimulus amplitudes
-    decimals: int of decimals to keep
+    return ramp_an.as_dict(
+        ramp_features,
+        [dict(sweep_number=sn) for sn in ramp_sweep_numbers]
+        )
 
-    Returns
-    -------
-    subthresh_min_amp: float min amplitude
-    min_amp_delta: min increment in the stimulus amplitude
-    """
 
-    amps_diff = np.round(np.diff(sorted(stim_amps)), decimals=decimals)
+def extract_cell_features(data_set, subthresh_min_amp=None):
 
-    amps_diff = amps_diff[amps_diff > 0]    # remove zeros, repeats are okay
-    amp_deltas = np.unique(amps_diff)   # unique nonzero deltas
-
-    if len(amp_deltas) == 0:
-        raise IndexError("All stimuli have identical amplitudes")
-
-    min_amp_delta = np.min(amp_deltas)
+    lu.log_pretty_header("Analyzing cell features:", level=2)
 
-    if len(amp_deltas) != 1:
-        logging.warning(
-            "Found multiple coarse long square amplitude step differences: %s.  Using: %f" % (str(amp_deltas), min_amp_delta))
+    cell_state = {"failed_fx": False, "fail_fx_message": None}
+    cell_features = {}
 
-    subthresh_min_amp = SUBTHRESHOLD_LONG_SQUARE_MIN_AMPS.get(min_amp_delta, None)
+    cell_features['long_squares'] = extract_cell_long_square_features(
+                                        data_set,
+                                        subthresh_min_amp,
+                                        cell_state=cell_state)
```

Comment:
As I mentioned earlier, I would not pass `cell_state` as an argument. Instead, I would simply make a wrapper to return an error message. Then, you could just append them to the list afterwords.